src/Xunit.StaFact.Desktop/Sdk/DispatcherSynchronizationContextAdapter.cs

```diff
@@ -41,5 +41,11 @@ internal override void Run(Func<Task> work)
             this.PumpTill(task);
             task.GetAwaiter().GetResult();
         }
+
+        internal override void Cleanup()
+        {
+            Dispatcher.CurrentDispatcher.InvokeShutdown();
```

Comment:
And from what I see in the the MSDN doc, [BeginInvokeShutdown](https://msdn.microsoft.com/en-us/library/system.windows.threading.dispatcher.begininvokeshutdown(v=vs.110).aspx) is the async version where as [InvokeShutdown](https://msdn.microsoft.com/en-us/library/system.windows.threading.dispatcher.invokeshutdown(v=vs.110).aspx) is the synchronous version